stdlib/public/core/Diffing.swift

```diff
@@ -0,0 +1,782 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2015 - 2019 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// MARK: Diff application to RangeReplaceableCollection
+
+extension RangeReplaceableCollection {
+  private static func fastApplicationEnumeration(
```

Comment:
This name needs an underscore, and it might work better as a method on `CollectionDifference`.
Additionally, the name of the function argument should be expanded. One random suggestion:

```suggestion
extension CollectionDifference {
  fileprivate func _fastEnumeratedApply(_ consume: (Change) -> Void) {...}
}
```